### DISCUSSION TITLE
Adding/Removing fields in Pocketbase Migrations

### DISCUSSION THREAD

**Discussion by starRMS** [pocketbase/pocketbase#3965] [Q&A] [answered]
So I am making a migration in Pocketbase that adds a field to the `users` model/table like the following:

```go
func init() {
	m.Register(func(db dbx.Builder) error {
		// Up queries...
		dao := daos.New(db)

		collection, err := dao.FindCollectionByNameOrId("users")
		if err != nil {
			return err
		}

		// API Rules
		collection.ListRule = nil
		collection.ViewRule = nil
		collection.CreateRule = nil
		collection.UpdateRule = nil
		collection.DeleteRule = nil

		collection.Indexes = types.JsonArray[string]{
			`CREATE UNIQUE INDEX idx_unique_user_nik ON users (some_field)`,
		}

		collection.Schema.AddField(&schema.SchemaField{
			Name:        "some_field",
			Type:        schema.FieldTypeText,
			Required:    true,
			Presentable: false,
			Unique:      true,
			Options: schema.TextOptions{
				Min:     types.Pointer(15),
				Max:     types.Pointer(16),
				Pattern: "^[0-9]+$",
			},
		})

		return nil
	}, func(db dbx.Builder) error {
		// Down queries...

		db.NewQuery(`ALTER TABLE users DROP COLUMN some_field`)

		return nil
	})
}
```

Now, how do I remove the field in the down query, I cannot find a `.RemoveField` method.

Am I currently doing the right things that is manually drop the column from the target table? Because I was thinking that Pocketbase itself keeps all schema in its database right? Don't I need to erase that too?

Thank you for any help.

**Comment by ganigeorgiev** [comment, marked as answer]
> I cannot find a .RemoveField method

Here it is - https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.20.0/models/schema#Schema.RemoveField.

> Because I was thinking that Pocketbase itself keeps all schema in its database right? Don't I need to erase that too?

No. If you call `dao.SaveCollection(collection)` PocketBase will update the collection model and will take care to sync the related records table changes.

Alternatively, note that you don't have to type migrations by hand and if you enable the automigrations via the `migratecmd.Config.Automigrate` field it will generate the migration files for you on every runtime collection change.

**Reply by starRMS** [reply]
Wow that is really cool about automigrations 🤯.

Sorry I should've read the pkg.go.dev more because I was only looking at the documentation at `https://pocketbase.io/docs`.

Thanks for the help!!! Cool project!
